### PR TITLE
HTTP headers are case insensitive handle them appropriately

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
An issue was reproduced when minio-js client functional
tests are setting lowercase HTTP headers, in our current
master branch we specifically look for canonical host header
which may be not necessarily true for all HTTP clients.
This leads to a perpetual hang on the *net.Conn*.

## Motivation and Context
This PR fixes regression caused by #6206 by handling the
case insensitivity.

## Regression
#6206 

## How Has This Been Tested?
Manually using mint tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/http/listener_test.go [hunk 1]
```diff
@@ -425,9 +425,11 @@ func TestHTTPListenerAccept(t *testing.T) {
 	}{
 		{[]string{"localhost:0"}, nil, "GET / HTTP/1.0\r\nHost: example.org\r\n\r\n", "200 OK\r\n", "GET / HTTP/1.0\r\n"},
 		{[]string{nonLoopBackIP + ":0"}, nil, "POST / HTTP/1.0\r\nHost: example.org\r\n\r\n", "200 OK\r\n", "POST / HTTP/1.0\r\n"},
+		{[]string{nonLoopBackIP + ":0"}, nil, "HEAD / HTTP/1.0\r\nhost: example.org\r\n\r\n", "200 OK\r\n", "HEAD / HTTP/1.0\r\n"},
 		{[]string{"127.0.0.1:0", nonLoopBackIP + ":0"}, nil, "CONNECT \r\nHost: www.example.org\r\n\r\n", "200 OK\r\n", "CONNECT \r\n"},
 		{[]string{"localhost:0"}, tlsConfig, "GET / HTTP/1.0\r\nHost: example.org\r\n\r\n", "200 OK\r\n", "GET / HTTP/1.0\r\n"},
 		{[]string{nonLoopBackIP + ":0"}, tlsConfig, "POST / HTTP/1.0\r\nHost: example.org\r\n\r\n", "200 OK\r\n", "POST / HTTP/1.0\r\n"},
+		{[]string{nonLoopBackIP + ":0"}, tlsConfig, "HEAD / HTTP/1.0\r\nhost: example.org\r\n\r\n", "200 OK\r\n", "HEAD / HTTP/1.0\r\n"},
```
Comment (nitisht) [inline cmd/http/listener_test.go:432]: can we also add a test case with lower case to make sure such issue doesn't happen again

Comment (harshavardhana) [reply cmd/http/listener_test.go:432]: That is exactly the test case added here @nitisht

Comment (nitisht) [reply cmd/http/listener_test.go:432]: oh right

**Review by nitisht — APPROVED** [review]

**Review by balamurugana — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#af946880-7b43-4a3e-9710-498baec33869) |
<a name="af946880-7b43-4a3e-9710-498baec33869"></a>
### 6390-40966a7/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31177
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0929bae41bca:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 1 minutes and 0 seconds
(2/13) Running aws-sdk-go tests ... done in 2 seconds
(3/13) Running aws-sdk-java tests ... done in 6 seconds
(4/13) Running aws-sdk-php tests ... done in 59 seconds
(5/13) Running aws-sdk-ruby tests ... done in 15 seconds
(6/13) Running mc tests ... done in 32 seconds
(7/13) Running minio-dotnet tests ... done in 5 minutes and 5 seconds
(8/13) Running minio-go tests ... done in 1 minutes and 44 seconds
(9/13) Running minio-java tests ... done in 3 minutes and 2 seconds
(10/13) Running minio-js tests ... done in 42 seconds
(11/13) Running minio-py tests ... FAILED in 1 hours 5 minutes and 14 seconds
{
  "duration": 3612510,
  "args": {
    "bucket_name": "minio-py-test-42a142a5-39b0-4a3e-8ced-d4049e577f04",
    "object_name": "2a0bce0e-1d49-46c3-a7f7-925cc7639533"
  },
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 600, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 384, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 2, in raise_from\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 380, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.5/http/client.py\", line 1197, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.5/http/client.py\", line 297, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.5/http/client.py\", line 266, in _read_status\n    raise RemoteDisconnected(\"Remote end closed connection without\"\nhttp.client.RemoteDisconnected: Remote end closed connection without response\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1427, in test_thread_safe\n    client.remove_object(bucket_name, object_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 942, in remove_object\n    object_name=object_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 1752, in _url_open\n    preload_content=preload_content)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/poolmanager.py\", line 322, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 638, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/util/retry.py\", line 398, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minikube', port=31177): Max retries exceeded with url: /minio-py-test-42a142a5-39b0-4a3e-8ced-d4049e577f04/2a0bce0e-1d49-46c3-a7f7-925cc7639533 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1800, in main\n    test_thread_safe(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1430, in test_thread_safe\n    raise Exception(err)\nException: HTTPConnectionPool(host='minikube', port=31177): Max retries exceeded with url: /minio-py-test-42a142a5-39b0-4a3e-8ced-d4049e577f04/2a0bce0e-1d49-46c3-a7f7-925cc7639533 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n",
  "function": "put_object(bucket_name, object_name, data, length, content_type, metadata)",
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "message": "HTTPConnectionPool(host='minikube', port=31177): Max retries exceeded with url: /minio-py-test-42a142a5-39b0-4a3e-8ced-d4049e577f04/2a0bce0e-1d49-46c3-a7f7-925cc7639533 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))"
}

Executed 10 out of 13 tests successfully.
```

**Pull request merged**